src/Middleware/CallableFactory.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Queue\Middleware;
+
+use Psr\Container\ContainerExceptionInterface;
+use Psr\Container\ContainerInterface;
+use ReflectionException;
+use ReflectionMethod;
+
+use function is_array;
+use function is_callable;
+use function is_string;
+
+/**
+ * @internal Create real callable listener from configuration.
+ */
+final class CallableFactory
+{
+    private ContainerInterface $container;
+
+    public function __construct(ContainerInterface $container)
+    {
+        $this->container = $container;
+    }
+
+    /**
+     * Create real callable listener from definition.
+     *
+     * @param mixed $definition Definition to create listener from.
+     *
+     * @throws InvalidCallableConfigurationException Failed to create listener.
+     * @throws ContainerExceptionInterface Error while retrieving the entry from container.
+     */
+    public function create(mixed $definition): callable
+    {
+        /** @var mixed */
+        $callable = $this->prepare($definition);
```

Comment:
This can be merged into this method. By using early returns it could be simpler.

Reply:
I've extracted the `prepare()` method. Does the result meet your expectations?

Reply:
Yes.